### PR TITLE
Add remote Host info to PageTransportErrorException

### PR DESCRIPTION
**Pull request by sanchitkashyap**
This would be useful to identify REMOTE_HOST_GONE for PTEE. For more info look -> https://github.com/prestodb/presto/pull/7691

### CODE REVIEW

File: presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java [hunk 1]
```diff
@@ -622,22 +623,22 @@ public PagesResponse handle(Request request, Response response)
...
                 if (!mediaTypeMatches(contentType, PRESTO_PAGES_TYPE)) {
-                    throw new PageTransportErrorException(format("Expected %s response from server but got %s", PRESTO_PAGES_TYPE, contentType));
+                    throw new PageTransportErrorException(fromUri(uri), format("Expected %s response from server but got %s", PRESTO_PAGES_TYPE, contentType));
                 }
 
-                String taskInstanceId = getTaskInstanceId(response);
-                long token = getToken(response);
-                long nextToken = getNextToken(response);
-                boolean complete = getComplete(response);
+                String taskInstanceId = getTaskInstanceId(response, uri);
```
Comment (losipiuk) [inline presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java:638]: do we need to pass `uri` to all those methods? We are catching `PageTransportErrorException` below anyway and add remote host information there. IMO that should be eough.

Comment (sanchitkashyap) [reply presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java:638]: I didn't do it that way, as that would require creating another constructor of `PageTransportErrorException` and pass remoteHost as null to its parent. I thought we would not want to do this (pass remoteHost as null for transport exceptions). I will change it if that is okay.

Comment (losipiuk) [reply presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java:638]: Yeah... Having `remoteHost` nullable is kinda ugly. So maybe we can live with this extra parameter passed to those helper methods. @electrum ?

@sanchitkashyap  Can you also comment on motivation for the change? The only final PTEE which is sent back to user is constructed in the `catch` clause below.
And we already have whole uri as part of an exception message there. Is that not enough?

File: presto-main/src/main/java/io/prestosql/operator/PageTransportErrorException.java [hunk 2]
```diff
@@ -13,20 +13,21 @@
...
 
-import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.HostAddress;
+import io.prestosql.spi.PrestoTransportException;
 
 import static io.prestosql.spi.StandardErrorCode.PAGE_TRANSPORT_ERROR;
 
 public class PageTransportErrorException
-        extends PrestoException
+        extends PrestoTransportException
```
Comment (findepi) [inline presto-main/src/main/java/io/prestosql/operator/PageTransportErrorException.java:22]: the change here seems not related to `Add remote Host info to PageTransportErrorException`

is it intentional?
make this into a separate commit

also cc @electrum

File: presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java (hunk 1 above)
Comment (sanchitkashyap) [reply presto-main/src/main/java/io/prestosql/operator/HttpPageBufferClient.java:638]: @losipiuk  i have updated the description to explain the motivation.

**Comment by losipiuk** [comment]
LGTM. I will let @electrum chime in to comment about making `PageTransportErrorException ` a `PrestoTransportException` now.

File: presto-main/src/main/java/io/prestosql/operator/PageTransportErrorException.java (hunk 2 above)
Comment (electrum) [reply presto-main/src/main/java/io/prestosql/operator/PageTransportErrorException.java:22]: This is needed and related since `PrestoTransportException` is the base class that contains the remote host, which is extracted in `Failures` to include as part of the error info. Given that `PageTransportTimeoutException` already extends it (and is the only subclass), I'm guessing we intended to do it for `PageTransportErrorException` originally.

**Comment by electrum** [comment]
Thanks!
